aptos-move/block-executor/src/unit_tests/mod.rs

```diff
@@ -681,75 +765,115 @@ fn scheduler_drain_idx() {
 fn finish_execution_wave() {
     // Wave won't be increased, because validation index is already 2, and finish_execution
     // tries to reduce it to 2.
-    let s = incarnation_one_scheduler(2);
+    let index_mapping = IndexMapping::new_unsharded(2);
+    let s = incarnation_one_scheduler(&index_mapping);
     assert!(matches!(
         s.finish_execution(1, 1, true),
         SchedulerTask::ValidationTask((1, 1), 0),
     ));
 
     // Here wave will increase, because validation index is reduced from 3 to 2.
-    let s = incarnation_one_scheduler(3);
+    let index_mapping = IndexMapping::new_unsharded(3);
+    let s = incarnation_one_scheduler(&index_mapping);
     assert!(matches!(
         s.finish_execution(1, 1, true),
         SchedulerTask::ValidationTask((1, 1), 1),
     ));
 
     // Here wave won't be increased, because we pass revalidate_suffix = false.
-    let s = incarnation_one_scheduler(3);
+    let index_mapping = IndexMapping::new_unsharded(3);
+    let s = incarnation_one_scheduler(&index_mapping);
     assert!(matches!(
         s.finish_execution(1, 1, false),
         SchedulerTask::ValidationTask((1, 1), 0),
     ));
 }
 
+#[test]
+fn finish_execution_wave_with_non_contiguous_indices() {
+    // Wave won't be increased, because validation index is already 2, and finish_execution
+    // tries to reduce it to 2.
+    let index_mapping = make_non_contiguous_index_mapping(2);
+    let s = incarnation_one_scheduler(&index_mapping);
+    assert!(matches!(
+        s.finish_execution(index_mapping.indices[1], 1, true),
+        SchedulerTask::ValidationTask((i, 1), 0) if i == index_mapping.indices[1],
+    ));
+
+    // Here wave will increase, because validation index is reduced from 3 to 2.
+    let index_mapping = make_non_contiguous_index_mapping(3);
+    let s = incarnation_one_scheduler(&index_mapping);
+    assert!(matches!(
+        s.finish_execution(index_mapping.indices[1], 1, true),
+        SchedulerTask::ValidationTask((i, 1), 1) if i == index_mapping.indices[1],
+    ));
+
+    // Here wave won't be increased, because we pass revalidate_suffix = false.
+    let index_mapping = make_non_contiguous_index_mapping(3);
+    let s = incarnation_one_scheduler(&index_mapping);
+    assert!(matches!(
+        s.finish_execution(index_mapping.indices[1], 1, false),
+        SchedulerTask::ValidationTask((i, 1), 0) if i == index_mapping.indices[1],
+    ));
+}
+
 #[test]
 fn rolling_commit_wave() {
-    let s = incarnation_one_scheduler(3);
+    rolling_commit_wave_main(IndexMapping::new_unsharded(3));
+}
+
+#[test]
+fn rolling_commit_wave_with_non_contiguous_indices() {
+    rolling_commit_wave_main(make_non_contiguous_index_mapping(3));
```

Comment:
I am wondering if we can refactor these tests avoid creating two tests for every test. Basically we want every test to run with 2 inputs. Take a look at `rtest` crate to see if that helps us achieve that.

Reply:
why avoiding two tests? I thought we may have some nice test runner that runs them in parallel.
And what's `rtest`? i didn't find such a crate...